std/string.d

```diff
@@ -3857,37 +3857,52 @@ unittest
     Returns:
         GC allocated string with tabs replaced with spaces
   +/
-S detab(S)(S s, size_t tabSize = 8) pure
-    if (isSomeString!S)
+auto detab(Range)(auto ref Range s, size_t tabSize = 8) pure
+    if ((isForwardRange!Range && isSomeChar!(ElementEncodingType!Range))
+        || __traits(compiles, StringTypeOf!Range))
+{
+    import std.array;
+    return detabber(cast(StringTypeOf!Range)s, tabSize).array;
+}
+
+///
+@trusted pure unittest
 {
     import std.array;
-    return detabber(s, tabSize).array;
+
+    assert(detab(" \n\tx", 9) == " \n         x");
 }
 
 /++
-    Replace each tab character in $(D r) with the number of spaces necessary
-    to align the following character at the next tab stop.
+    Replace each tab character in $(D input) with the number of spaces
+    necessary to align the following character at the next tab stop.
 
     Params:
-        r = string or forward range
+        input = string or forward range
         tabSize = distance between tab stops
 
     Returns:
         lazy forward range with tabs replaced with spaces
   +/
-auto detabber(Range)(Range r, size_t tabSize = 8)
+auto detabber(Range)(Range input, size_t tabSize = 8)
     if (isForwardRange!Range && isSomeChar!(ElementEncodingType!Range))
 {
     import std.uni : lineSep, paraSep, nelSep;
     import std.utf : codeUnitLimit, decodeFront;
 
     assert(tabSize > 0);
-    alias C = Unqual!(ElementEncodingType!Range);
+
+    static if (__traits(compiles, StringTypeOf!Range))
+        StringTypeOf!Range r = input;
+    else
+        alias r = input;
+
+    alias C = Unqual!(ElementEncodingType!(typeof(r)));
```

Comment:
You can remove all the changes to the original detabber function because StringTypeOf  is now longer in the constraint.